File: Web-Ecommerce/Front/shared/types/order.ts
import type { Address } from './user'
import type { PageQuery } from './index'

export interface OrderItem {
  id: number
  orderId: number
  productId: number
  productName: string
  productImage: string
  skuId: number
  specDesc: string
  quantity: number
  price: number
  isReviewed?: boolean
  hasFollowUp?: boolean
}

export interface Order {
  id: number
  orderNo: string
  userId: number
  addressId: number
  address: Address
  totalAmount: number
  couponIds?: string
  couponDiscount?: number
  discountAmount?: number
  payAmount?: number
  couponName?: string
  status: number
  statusText: string
  payMethod?: string
  payTime: string
  shipTime?: string
  dealTime?: string
  createTime: string
  updateTime?: string
  addressModified?: number
  reviewCount?: number
  items: OrderItem[]
  // Refund fields
  refundType?: number
  refundReason?: string
  refundReasonText?: string
  refundDesc?: string
  refundAmount?: number
  refundItemIds?: string
  refundStatus?: number
  refundStatusText?: string
  refundRejectReason?: string
  refundApplyTime?: string
  refundDealTime?: string
  refundItems?: OrderItem[]
}

export interface OrderQuery extends PageQuery {
  status?: number
  userId?: number
  reviewFilter?: string  // 'pending' | 'followup' | 'reviewed'
  hasRefund?: boolean
}

export interface CreateOrderForm {
  addressId: number
  cartItemIds?: number[]
  productId?: number
  skuId?: number
  quantity?: number
  remark?: string
  userCouponIds?: number[]
}

export interface PaymentForm {
  orderId: number
  payMethod: string
}

export interface PayIntent {
  qrToken: string
  orderNo: string
  amount: number
  payMethod: string
}

export interface PayStatus {
  status: 'WAITING_SCAN' | 'SCANNED' | 'PAID' | 'NONE'
  scanned: boolean
  payMethod: string | null
}

File: Web-Ecommerce/Back/src/main/java/com/ecommerce/entity/Order.java
package com.ecommerce.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("`order`")
public class Order {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String orderNo;
    private Long userId;
    private Long addressId;
    private BigDecimal totalAmount;
    private String couponIds;
    private BigDecimal couponDiscount;
    private BigDecimal discountAmount;
    private BigDecimal payAmount;
    private Integer status;
    private String remark;
    private Integer refundType;
    private String refundReason;
    private String refundDesc;
    private java.math.BigDecimal refundAmount;
    private String refundItemIds;
    private Integer refundStatus;
    private String refundRejectReason;
    private LocalDateTime refundApplyTime;
    private LocalDateTime refundDealTime;
    private LocalDateTime payTime;
    private LocalDateTime dealTime;
    private Integer addressModified;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private String statusText;
    @TableField(exist = false)
    private List<OrderItem> items;
    @TableField(exist = false)
    private Object address;
    @TableField(exist = false)
    private Long reviewCount;
    @TableField(exist = false)
    private String couponName;
    @TableField(exist = false)
    private String refundReasonText;
    @TableField(exist = false)
    private String refundStatusText;
    @TableField(exist = false)
    private java.util.List<OrderItem> refundItems;
}
